### PR TITLE
Python: modernise cls self argument name checks

### PR DESCRIPTION
**Pull request by RasmusWL**
One thing I'm not sure about is the fact that I deprecated the old `class ZopeInterfaceMethod extends PyFunctionObject`,  but I thought it was the easy way to remember it by doing it now!

### CODE REVIEW

File: python/ql/src/Functions/NonCls.ql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * @name First parameter of a class method is not named 'cls'
+ * @name First argument of a class method is not named 'cls'
```
Comment (taus-semmle) [inline python/ql/src/Functions/NonCls.ql]: `parameter` was the right word here, not `argument`. https://docs.python.org/3/faq/programming.html#faq-argument-vs-parameter

File: python/ql/src/Functions/NonCls.ql [hunk 2]
```diff
@@ -23,13 +23,13 @@ predicate first_arg_cls(Function f) {
...
+    exists(ClassValue c | c.getScope() = f.getScope() and c.getASuperType() = ClassValue::type())
 }
 
 predicate classmethod_decorators_only(Function f) {
-    forall(Expr decorator |
-        decorator = f.getADecorator() |
-            ((Name) decorator).getId() = "classmethod")
+  forall(Expr decorator | decorator = f.getADecorator() |
+      decorator.(Name).getId() = "classmethod"
+  )
```
Comment (taus-semmle) [inline python/ql/src/Functions/NonCls.ql]: Oh no! The dreaded two-space indentation! :slightly_smiling_face:

File: python/ql/src/Functions/NonCls.ql (hunk 1 above)
Comment (taus-semmle) [reply python/ql/src/Functions/NonCls.ql]: This of course means we were using the _wrong_ term in `NonSelf.ql`...

File: python/ql/test/query-tests/Functions/general/argument_names.py [hunk 3]
```diff
@@ -66,3 +74,46 @@ def s_smethod2(ok):
...
+class Acceptable2(object):
+    def _func(f):
+        return f
+
+    @_func
+    def meth(self):
+        pass
+
+# Handling methods defined in a different scope than the class it belongs to,
+# get's problmematic since we need to show the full-path from method definition
```
Comment (taus-semmle) [inline python/ql/test/query-tests/Functions/general/argument_names.py]: Typo: `gets`

File: python/ql/src/Functions/NonCls.ql (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/Functions/NonCls.ql]: and we're also using the wrong term in the following 2 lines.

PEP8 also needs to be updated it seems, as it [says](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments):

> ### Function and Method Arguments
> Always use self for the first argument to instance methods.
>
> Always use cls for the first argument to class methods.

File: python/ql/test/query-tests/Functions/general/argument_names.py (hunk 3 above)
Comment (taus-semmle) [reply python/ql/test/query-tests/Functions/general/argument_names.py]: Oh, and `problematic`.

**Comment by RasmusWL** [comment]
Fix the things from review + rebased on newest master... so now it should hopefully all be good :tada:

**Comment by RasmusWL** [comment]
Forgot to update the filename in the `.expected` test files. now tests should actually pass :tada:

**Pull request merged**